src/main/java/com/google/firebase/auth/UserRecord.java

```diff
@@ -177,6 +191,14 @@ private static void checkEmail(String email) {
     checkArgument(email.matches("^[^@]+@[^@]+$"));
   }
 
+  private static void checkPhoneNumber(String phoneNumber) {
+    // Phone number verification is very lax here. Backend will enforce E.164 spec compliance, and
+    // normalize accordingly.
+    checkArgument(!Strings.isNullOrEmpty(phoneNumber), "phone number cannot be null or empty");
+    checkState(phoneNumber.matches("^\\+.*[a-zA-Z0-9].*$"),
```

Comment:
If I read this correctly,  this merely verifies that the phone number starts with '+' and contains at least one 'normal' character. So +&&&a&&& might be a valid number? If that is our level of verification, then we might as well let the server handles it.